torchvision/transforms/v2/functional/_utils.py

```diff
@@ -101,78 +113,44 @@ def _get_kernel(dispatcher, input_type):
             elif cls in registry:
                 return registry[cls]
 
-        # Note that in the future we are not going to return a noop here, but rather raise the error below
+    if allow_passthrough:
         return _noop
 
-    raise TypeError(
-        f"Dispatcher {dispatcher} supports inputs of type torch.Tensor, PIL.Image.Image, "
-        f"and subclasses of torchvision.datapoints.Datapoint, "
-        f"but got {input_type} instead."
-    )
-
+    supported_datapoint_types = registry.keys() - {torch.Tensor, PIL.Image.Image}
+    builtin_datapoint_types = supported_datapoint_types & _BUILTIN_DATAPOINT_TYPES
+    custom_datapoint_types = supported_datapoint_types - builtin_datapoint_types
 
-# Everything below this block is stuff that we need right now, since it looks like we need to release in an intermediate
-# stage. See https://github.com/pytorch/vision/pull/7747#issuecomment-1661698450 for details.
+    builtin_datapoint_type_names = ", ".join(f"datapoints.{t.__name__}" for t in builtin_datapoint_types)
+    custom_datapoint_types_names = ", ".join(str(t) for t in custom_datapoint_types)
 
+    supported_type_names = "torch.Tensor"
+    supported_type_names += ", PIL.Image.Image, " if PIL.Image.Image in registry else " "
+    supported_type_names += f"and subclasses of the builtin {builtin_datapoint_type_names}"
+    if custom_datapoint_types:
+        supported_type_names += f", and the custom datapoint types {custom_datapoint_types_names}"
 
-# In the future, the default behavior will be to error on unsupported types in dispatchers. The noop behavior that we
-# need for transforms will be handled by _get_kernel rather than actually registering no-ops on the dispatcher.
-# Finally, the use case of preventing users from registering kernels for our builtin types will be handled inside
-# register_kernel.
-def _register_explicit_noop(*datapoints_classes, warn_passthrough=False):
-    """
-    Although this looks redundant with the no-op behavior of _get_kernel, this explicit registration prevents users
-    from registering kernels for builtin datapoints on builtin dispatchers that rely on the no-op behavior.
-
-    For example, without explicit no-op registration the following would be valid user code:
-
-    .. code::
-        from torchvision.transforms.v2 import functional as F
+    raise TypeError(
+        f"Dispatcher F.{dispatcher.__name__} supports inputs of type {supported_type_names}, "
+        f"but got {input_type} instead."
+    )
 
-        @F.register_kernel(F.adjust_brightness, datapoints.BoundingBox)
-        def lol(...):
-            ...
-    """
 
+def _register_temporary_passthrough_kernels_internal(*datapoints_classes):
```

Comment:
we should align the uses of "passthrough" and "noop".

It's OK to use passthrough for transforms and noop for functionals but here (in the functionals) we're using both terms and we should probably just use "noop" (or just "passthrough")